language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AssignmentStatementNodeContext.java

```diff
@@ -106,24 +106,38 @@ public void sort(BallerinaCompletionContext context, AssignmentStatementNode nod
 
         TypeSymbol symbol = typeSymbolAtCursor.get();
         completionItems.forEach(completionItem -> {
-            int rank = SortingUtil.toRank(completionItem, 1);
+            int rank = SortingUtil.toRank(completionItem, 2);
 
-            // If a completion item is a symbol and is assignable to the variable at left hand side, 
-            // this assigns the highest rank to such variables and methods.
+            // If a completion item is a symbol and is assignable to the variable at left hand side,
+            // this assigns the highest rank to such variables and then the next highest to functions.
             if (completionItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
-                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) completionItem;
 
-                Optional<TypeSymbol> completionItemType =
-                        SymbolUtil.getTypeDescriptor(symbolCompletionItem.getSymbol());
-                if (completionItemType.isPresent() && completionItemType.get() instanceof FunctionTypeSymbol) {
-                    completionItemType = ((FunctionTypeSymbol) completionItemType.get()).returnTypeDescriptor();
+                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) completionItem;
+                Optional<TypeSymbol> evalTypeSymbol = SymbolUtil.getTypeDescriptor(symbolCompletionItem.getSymbol());
+                CompletionItemKind completionItemKind = completionItem.getCompletionItem().getKind();
+
+                if (evalTypeSymbol.isPresent()) {
+
+                    switch (completionItemKind) {
+                        case Variable:
+                            if (evalTypeSymbol.get().assignableTo(symbol)) {
+                                rank = 1;
+                            }
+                            break;
+                        case Method:
+                        case Function:
+                            if (symbolCompletionItem.getSymbol() instanceof FunctionSymbol) {
+                                Optional<TypeSymbol> returnType = ((FunctionSymbol) symbolCompletionItem.getSymbol())
```

Comment:
Can you check what happens when the `returnType` is a `type reference`? i.e: Try adding a user defined type as the return type of a function and see. If what you get is a `type reference`, then use `CommonUtil.getRawType()` method to get the actual type.

If `assignableTo()` call internally handles the above scenario, no change is required as well.

Reply:
This is internally handled